### PR TITLE
[Popover] react-popper 1.0.0

### LINKED ISSUE #2530: use react-popper 1.0.0
it's just got so many great features! lets us remove all wrapper elements.

also will be able to remove a bunch of special CSS for handling popover wrappers.
- button groups
- trees

### PR DESCRIPTION
**Pull request by giladgray**
#### Changes proposed in this pull request:

- upgrade to `react-popper@1.0.0` which features a totally new API using `children` functions

### CODE REVIEW

File: packages/core/src/components/tooltip/tooltip.tsx [hunk 1]
```diff
@@ -95,17 +95,11 @@ export interface ITooltipProps extends IProps, IIntentProps, IOverlayLifecyclePr
      */
     position?: Position | "auto";
 
-    /**
-     * The name of the HTML tag to use when rendering the popover target wrapper element (`Classes.POPOVER_WRAPPER`).
-     * @default "span"
-     */
-    rootElementTag?: string;
```
Comment (invliD) [inline packages/core/src/components/tooltip/tooltip.tsx]: why not let tooltips provide a `wrapperTagName`?

Comment (giladgray) [reply packages/core/src/components/tooltip/tooltip.tsx]: i have a follow up PR to introduce `IPopoverSharedProps` that will resolve this.

Comment (giladgray) [reply packages/core/src/components/tooltip/tooltip.tsx]: maybe i'll just open that now as there's a merge conflict here

**Comment by giladgray** [comment]
opened #2617 which will reduce the diff here if merged first

**Pull request merged**